clients/ts/FunctionalTests/ts/WebSocketTests.ts

```diff
@@ -1,33 +1,37 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+import { WebSocketWrapper } from "@aspnet/signalr/dist/esm/Polyfills";
 import { ECHOENDPOINT_URL } from "./Common";
 
 // On slower CI machines, these tests sometimes take longer than 5s
 jasmine.DEFAULT_TIMEOUT_INTERVAL = 10 * 1000;
 
-if (typeof WebSocket !== "undefined") {
-    describe("WebSockets", () => {
-        it("can be used to connect to SignalR", (done) => {
-            const message = "message";
+describe("WebSockets", () => {
+    it("can be used to connect to SignalR", (done) => {
+        const message = "message";
 
-            const webSocket = new WebSocket(ECHOENDPOINT_URL.replace(/^http/, "ws"));
+        let webSocket: WebSocket;
+        if (typeof WebSocket !== "undefined") {
```

Comment:
We should gracefully handle the case of running the tests in a browser that doesn't have WebSockets. Right now that will fall through this condition to the else and you'll have a Bad Problem(TM).

Change this condition to `if(window)` and then test `typeof WebSocket` inside the if and if it's undefined just no-op the test.